### ISSUE TITLE
Thread names

### ISSUE THREAD

**Issue by CookiePLMonster** [OpenRCT2/OpenRCT2#14713]
Right now, all OpenRCT2 threads are unnamed. This makes debugging harder than it could be and potentially loses some information in crashdumps (if names were to be included in them).

![image](https://user-images.githubusercontent.com/7947461/119187347-84fe4e80-ba79-11eb-96c8-6576450e7bba.png)

Consider using `pthread_setname_np` on POSIX platforms and [SetThreadName](https://docs.microsoft.com/en-us/visualstudio/debugger/how-to-set-a-thread-name-in-native-code?view=vs-2019) on Windows.

On high level, it'd be as simple as calling a `void SetThreadName(const utf8* name);` at the top of a newly started thread. Naming from another thread is technically possible, but I've encountered issues with it in the past while "self-naming" was always reliable.

**Comment by ZehMatt** [comment]
We have to be a bit careful about that on Windows, naming a thread is something that didn't exist prior to Windows 10. On earlier versions it used a special exception to notify the debugger about the name but that did not name the thread, it just informs the debugger about it.

**Comment by CookiePLMonster** [comment]
Indeed, so for the sake of crashdumps only the Windows 10 variation would be meaningful. I guess both can be supported depending on the OS the user runs on.
